graphix/graphsim/basegraphstate.py

```diff
@@ -19,12 +20,22 @@
 
     RUSTWORKX_INSTALLED = True
 except ModuleNotFoundError:
+    # !!! INVALID !!!
+    # We cannot define "dynamic" type hints
     rx = None
+    # PyGraph is not regarded as a type hint here
     PyGraph = None
```

Comment:
`rx = None` can be removed, perhaps?
`PyGraph` part - not sure yet how best to proceed. we can, in principle, require `rustworkx` instead of `networkx`, if this structure become too much of burden for maintaining/cleaning up. What is your thoughts @EarlMilktea ?

Reply:
I personally prefer to make `import` itself optional: optional module is not imported automatically, and if users import without installing necessary optional deps., exception is raised.

With this design, we don't have to keep going after `ImportError` .

Reply:
I see, does that mean rustworkx is `import`'ed only when `RxGraphState` is instantiated?

Reply:
I misunderstood a little bit...
As this file is for the common base class, we cannot skip importing it.
Let me discuss workarounds later.

Reply:
The situation is not so easy as I expected...

@shinich1 Could you review and merge (if OK) this PR first?

I may need to carefully examine the codes to fix the problem above.

Reply:
@EarlMilktea do we want to keep the comments here (`!!! invalid ...`) or is it cleaner to reomve this part and put in an issue?

This PR looks good overall, let me take a look again later. I also ask @thierry-martinez to have a quick look through.

Reply:
May better to open a new issue.

Reply:
Sounds good! 
please remove these lines before merging, and put in a new issue.

Reply:
Opened #171 .